jscomp/ml/variant_coercion.ml

```diff
@@ -6,15 +6,28 @@ let can_coerce_path (path : Path.t) =
   || Path.same path Predef.path_int
   || Path.same path Predef.path_float
 
-let can_coerce_variant ~(path : Path.t)
+let check_paths_same p1 p2 target_path =
+  Path.same p1 target_path && Path.same p2 target_path
+
+let can_coerce_variant ~(path : Path.t) ~unboxed
```

Comment:
I think the function's name should indicate that this function is for coercing a variant `to` a type.
And the for_all commented or refactored so the high-level algorithm is clear:
- every case of the variant has the same runtime representation as the target type.

This allows to treat `string` and `float` at the same time as both conform to that algorithm.

Reply:
Yes, good catch! I'll clean these things up a bit.

Reply:
Merge away when done.

Reply:
@cristianoc please look at https://github.com/rescript-lang/rescript-compiler/pull/6441/commits/95be068eb72e3e08ad0280bf10708b4c8212fc55 to see if the refactor is clear enough.

Reply:
Looks great